### PR TITLE
Use string equality operator for comparison; Simplify boolean expressions.

### PR DESCRIPTION
**Pull request by endJunction**
See each commit's message for details.

Originally only the string comparison was suggested, but that was sometimes used in unnecessary complicated boolean expressions.

### CODE REVIEW

File: Applications/FileIO/TetGenInterface.cpp [hunk 1]
```diff
@@ -416,10 +416,7 @@ bool TetGenInterface::parseElementsFileHeader(std::string &line,
     pos_end = line.find_first_of(" \t\n", pos_beg);
     if (pos_end == std::string::npos)
         pos_end = line.size();
-    if ((line.substr(pos_beg, pos_end - pos_beg)).compare("1") == 0)
-        region_attribute = true;
-    else
-        region_attribute = false;
+    region_attribute = (line.substr(pos_beg, pos_end - pos_beg)) == "1";
```
Comment (chleh) [inline Applications/FileIO/TetGenInterface.cpp]: can be simplified.

File: GeoLib/AnalyticalGeometry.cpp [hunk 2]
```diff
@@ -135,10 +135,7 @@ bool lineSegmentIntersect(GeoLib::LineSegment const& s0,
                                             std::size_t i)
     {
         // check if p is located at v=(a,b): (ap = t*v, t in [0,1])
-        if (0.0 <= ap[i] / v[i] && ap[i] / v[i] <= 1.0) {
-            return true;
-        }
-        return false;
+        return 0.0 <= ap[i] / v[i] && ap[i] / v[i] <= 1.0;
```
Comment (chleh) [inline GeoLib/AnalyticalGeometry.cpp]: Not this PR's change, but what if `v[i]` is zero?

File: GeoLib/GEOObjects.cpp [hunk 3]
```diff
@@ -150,7 +150,7 @@ bool GEOObjects::appendPolylineVec(const std::vector<Polyline*>& polylines,
     std::size_t idx (0);
     bool nfound (true);
     for (idx = 0; idx < _ply_vecs.size() && nfound; idx++)
-        if ( (_ply_vecs[idx]->getName()).compare (name) == 0 )
+        if ((_ply_vecs[idx]->getName()) == name)
```
Comment (chleh) [inline GeoLib/GEOObjects.cpp]: too many (())

File: GeoLib/GEOObjects.cpp [hunk 4]
```diff
@@ -222,7 +222,7 @@ bool GEOObjects::appendSurfaceVec(const std::vector<Surface*>& surfaces,
     std::size_t idx (0);
     bool nfound (true);
     for (idx = 0; idx < _sfc_vecs.size() && nfound; idx++)
-        if ( (_sfc_vecs[idx]->getName()).compare (name) == 0 )
+        if ((_sfc_vecs[idx]->getName()) == name)
```
Comment (chleh) [inline GeoLib/GEOObjects.cpp]: (()), also elsewhere.

File: GeoLib/SensorData.cpp [hunk 5]
```diff
@@ -168,10 +168,11 @@ std::string SensorData::convertSensorDataType2String(SensorDataType t)
 
 SensorDataType SensorData::convertString2SensorDataType(const std::string &s)
 {
-    if ((s.compare("Evaporation")==0) || (s.compare("EVAPORATION")==0)) return SensorDataType::EVAPORATION;
-    if ((s.compare("Precipitation") == 0) || (s.compare("PRECIPITATION") == 0))
+    if ((s == "Evaporation") || (s == "EVAPORATION"))
```
Comment (chleh) [inline GeoLib/SensorData.cpp]: less parentheses needed.

File: MeshLib/MeshEnums.cpp [hunk 6]
```diff
@@ -60,21 +60,21 @@ const std::string MeshElemType2StringShort(const MeshElemType t)
 
 MeshElemType String2MeshElemType(const std::string &s)
 {
-    if ((s.compare("point") == 0) || (s.compare("Point") == 0))
+    if ((s == "point") || (s == "Point"))
```
Comment (chleh) [inline MeshLib/MeshEnums.cpp]: less parens.

File: GeoLib/Polygon.cpp [hunk 7]
```diff
@@ -183,9 +183,9 @@ bool Polygon::containsSegment(GeoLib::LineSegment const& segment) const
         if (!isPntInPolygon(GeoLib::Point(0.5*(s[k][0]+s[k+1][0]), 0.5*(s[k][1]+s[k+1][1]), 0.5*(s[k][2]+s[k+1][2]))))
         return false;
     }
-    if (!isPntInPolygon(GeoLib::Point(0.5*(s[0][0]+b[0]), 0.5*(s[0][1]+b[1]), 0.5*(s[0][2]+b[2]))))
-        return false;
-    return true;
+    return !!isPntInPolygon(GeoLib::Point(0.5 * (s[0][0] + b[0]),
```
Comment (TomFischer) [inline GeoLib/Polygon.cpp]: I think `!!` could be removed.

File: Applications/FileIO/Gmsh/GmshReader.cpp [hunk 8]
```diff
@@ -167,12 +167,14 @@ MeshLib::Mesh* readGMSHMesh(std::string const& fname)
...
     getline(in, line); // version-number file-type data-size
-    if (line.substr(0,3).compare("2.2") != 0) {
+    if (line.substr(0, 3) != "2.2")
+    {
         WARN("Wrong gmsh file format version.");
         return nullptr;
     }
 
-    if (line.substr(4,1).compare("0") != 0) {
+    if (line.substr(4, 1) != "0")
```
Comment (endJunction) [inline Applications/FileIO/Gmsh/GmshReader.cpp]: Found only two more of substr(*,1). Fixed.

File: Applications/FileIO/TetGenInterface.cpp [hunk 9]
```diff
@@ -107,7 +107,7 @@ std::size_t TetGenInterface::getNFacets(std::ifstream &input)
         auto it = fields.begin();
         const auto nFacets(BaseLib::str2number<std::size_t>(*it));
         if (fields.size() > 1)
-            _boundary_markers = (BaseLib::str2number<std::size_t> (*(++it)) == 0) ? false : true;
+            _boundary_markers = BaseLib::str2number<std::size_t>(*(++it)) != 0;
```
Comment (endJunction) [inline Applications/FileIO/TetGenInterface.cpp]: Not sure. It could be an uninitialized value...

**Comment by endJunction** [comment]
Only the pointed `((` fixed, but there are many more in the code.
Accidently found out that QString.compare("abc") is also used often; this change will come later.

**Pull request merged**